Cargo.toml

```diff
@@ -39,10 +39,10 @@ colored = "2"
 ui_test = "0.21.1"
 rustc_version = "0.4"
 # Features chosen to match those required by env_logger, to avoid rebuilds
-regex = { version = "1.5.5", default-features = false, features = ["perf", "std"] }
+regex = { version = "1.9.6", default-features = false, features = ["perf", "std"] }
 lazy_static = "1.4.0"
 # Require a version of serde without intransparent unreproducible binary blobs.
-serde = { version = "1.0.185", features = ["derive"] }
+serde = { version = "1.0.188", features = ["derive"] }
```

Comment:
Why this change? This is still the same semver bracket.

(Also applies for some of the other toml file changes)

Reply:
What is the criterion? Specify the latest version available at the time the dependency is added and keep it frozen forever in `Cargo.toml` and only update it in `Cargo.lock`?

Reply:
That's what we did so far. If we want something more principled it should be the minimum number that works (and we should have a CI job that uses minimal-version resolution -- which might nor work since some widely used ecosystem crates historically didn't care for working builds under minimal versions).

Reply:
So be it